### PR TITLE
Amp first party data

### PR DESCRIPTION
**Pull request by BraslavskiyAndrey**

### CODE REVIEW

File: src/main/java/org/prebid/server/auction/AmpRequestFactory.java [hunk 1]
```diff
@@ -221,37 +227,64 @@ private BidRequest overrideParameters(BidRequest bidRequest, HttpServerRequest r
             }
         }
 
-        final Site updatedSite = overrideSite(bidRequest.getSite(), request);
-        final Imp updatedImp = overrideImp(bidRequest.getImp().get(0), request);
+        final String requestTargeting = request.getParam(TARGETING_REQUEST_PARAM);
+        final Targeting targeting = parseTargeting(requestTargeting);
+        final ObjectNode targetingNode = (ObjectNode) readTargeting(requestTargeting);
```
Comment (schernysh) [inline src/main/java/org/prebid/server/auction/AmpRequestFactory.java:232]: This cast is worrisome - why not check if parsed `JsonNode` is object and make cast in `readTargeting`?

File: src/main/java/org/prebid/server/auction/AmpRequestFactory.java [hunk 2]
```diff
@@ -221,37 +227,64 @@ private BidRequest overrideParameters(BidRequest bidRequest, HttpServerRequest r
...
         } else {
             result = bidRequest;
         }
         return result;
     }
 
-    private Site overrideSite(Site site, HttpServerRequest request) {
+    private JsonNode readTargeting(String requestTargeting) {
+        try {
+            return requestTargeting != null ? mapper.mapper().readTree(requestTargeting) : null;
```
Comment (schernysh) [inline src/main/java/org/prebid/server/auction/AmpRequestFactory.java:260]: `requestTargeting` looks to be the same value as input parameter in `parseTargeting`. However here it is not decoded prior to parsing. Is it intentional difference?

Comment (BraslavskiyAndrey) [reply src/main/java/org/prebid/server/auction/AmpRequestFactory.java:260]: It is mistake, fixed

File: src/main/java/org/prebid/server/auction/AmpRequestFactory.java (hunk 1 above)
Comment (BraslavskiyAndrey) [reply src/main/java/org/prebid/server/auction/AmpRequestFactory.java:232]: fixed

**Pull request merged**